cmake/BLTMacros.cmake

```diff
@@ -380,6 +400,175 @@ macro(blt_register_library)
 
 endmacro(blt_register_library)
 
+##------------------------------------------------------------------------------
+## blt_patch_target( NAME <targetname>
+##                   DEPENDS_ON [dep1 [dep2 ...]]
+##                   INCLUDES [include1 [include2 ...]]
+##                   LIBRARIES [lib1 [lib2 ...]]
+##                   TREAT_INCLUDES_AS_SYSTEM [ON|OFF]
+##                   FORTRAN_MODULES [ path1 [ path2 ..]]
+##                   COMPILE_FLAGS [ flag1 [ flag2 ..]]
+##                   LINK_FLAGS [ flag1 [ flag2 ..]]
+##                   DEFINES [def1 [def2 ...]] )
+##
+## Modifies an existing CMake target - sets PUBLIC visibility except for INTERFACE
+## libraries, which use INTERFACE visibility
+##------------------------------------------------------------------------------
+macro(blt_patch_target)
+    set(singleValueArgs NAME TREAT_INCLUDES_AS_SYSTEM)
+    set(multiValueArgs INCLUDES 
+                       DEPENDS_ON
+                       LIBRARIES
+                       FORTRAN_MODULES
+                       COMPILE_FLAGS
+                       LINK_FLAGS
+                       DEFINES )
+
+    ## parse the arguments
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # Input checks
+    if( "${arg_NAME}" STREQUAL "" )
+        message(FATAL_ERROR "blt_patch_target() must be called with argument NAME <name>")
+    endif()
+
+    if (NOT TARGET ${arg_NAME})
+        message(FATAL_ERROR "blt_patch_target() NAME argument must be a native CMake target")
+    endif()
+
+    # Default to public scope, unless it's an interface library
+    set(_scope PUBLIC)
+    get_target_property(_target_type ${arg_NAME} TYPE)
+    if(${_target_type} STREQUAL "INTERFACE_LIBRARY")
```

Comment:
Please change this to `"${_target_type}"`, I have been bit too many times by CMake's evaluation of if statements with strings.